tictactoe.rb

```diff
@@ -1,184 +1,185 @@
+			else
+				first = 3
+			end
+
+			case column
+			when "left"
+				second = 0
+			when "middle"
+				second = 1
+			when "right"
+				second = 2
+			else
+				second = 3
+			end
+				move_idx = [first, second]
+			return move_idx
 		end
-			move_idx = [first, second]
-		return move_idx
-	end
 
-	def valid_move?(row, column)
-
-	  if row == 3 || column == 3
-	  	puts "Please type: top middle bottom, with: left middle right"
-	  	return false
-	  elsif @board[row][column] != "-"
-			puts "You cannot overwrite another move. Try Again you fucking idiot."
-			return false
-	  else
-	  	return true
-	  end
-	end
+		def valid_move?(row, column)
+
+		  if row == 3 || column == 3
+		  	puts "Please type: top middle bottom, with: left middle right"
+		  	return false
+		  elsif @board[row][column] != "-"
+				puts "You cannot overwrite another move. Try Again you fucking idiot."
+				return false
+		  else
+		  	return true
+		  end
+		end
 
-	def winner
-		i = 0
-		winner_logic = false
-		while i < @board.length
-			if (@board[i][0] == @board[i][1]) && (@board[i][1] == @board[i][2]) && (@board[i][0] != "-")
-				# puts "#{@board[i][0]} is the winner!!!"
-				winner_logic = true
+		def winner
+			i = 0
+			winner_logic = false
+			while i < @board.length
+				if (@board[i][0] == @board[i][1]) && (@board[i][1] == @board[i][2]) && (@board[i][0] != "-")
+					# puts "#{@board[i][0]} is the winner!!!"
+					winner_logic = true
+				end
+				i += 1
 			end
-			i += 1
-		end
-		j = 0
-		while j < @board.length
-			if @board[0][j] == @board[1][j] && @board[1][j] == @board[2][j] && @board[0][j] != "-"
-				# puts "#{@board[0][j]} is the winner!!!"
+			j = 0
+			while j < @board.length
+				if @board[0][j] == @board[1][j] && @board[1][j] == @board[2][j] && @board[0][j] != "-"
+					# puts "#{@board[0][j]} is the winner!!!"
+					winner_logic = true
+				end
+				j += 1
+			end
+			if @board[0][0] == @board[1][1] && @board[1][1] == @board[2][2] && @board[0][0] != "-"
+				# puts "#{@board[0][0]} is the winner!!!"
+				winner_logic = true
+			elsif @board[2][0] == @board[1][1] && @board[1][1] == @board[0][2] && @board[2][0] != "-"
+				# puts "#{@board[2][0]} is the winner!!!"
 				winner_logic = true
 			end
-			j += 1
+			return winner_logic
 		end
-		if @board[0][0] == @board[1][1] && @board[1][1] == @board[2][2] && @board[0][0] != "-"
-			# puts "#{@board[0][0]} is the winner!!!"
-			winner_logic = true
-		elsif @board[2][0] == @board[1][1] && @board[1][1] == @board[0][2] && @board[2][0] != "-"
-			# puts "#{@board[2][0]} is the winner!!!"
-			winner_logic = true
-		end
-		return winner_logic
-	end
 
 
-	def board_not_full
-		logic = @board.any? do |x|
-			x.include?("-")
+		def board_not_full
+			logic = @board.any? do |x|
+				x.include?("-")
 
+			end
 		end
-	end
 
-	def stale_mate
-		puts "No moves left"
-		puts "D R A W"
+		def stale_mate
+			puts "No moves left"
+			puts "D R A W"
+		end
 	end
-end
 
-new_board = Board.new 
+	new_board = Board.new 
```

Comment:
You should have a file probably `Game.rb`. That initializing the new game and board. It appears this is just kind of thrown down here to set up a new board and game but try to organize this section of code a bit better.